vra/resource_deployment.go

```diff
@@ -905,7 +907,26 @@ func runDeploymentUpdateAction(ctx context.Context, d *schema.ResourceData, apiC
 			// to the type defined in the schema.
 			inputs, err = getCatalogItemInputsByType(apiClient, catalogItemID, catalogItemVersion, v)
 			if err != nil {
-				return err
+
+				// If the catalog item version is no longer available,
+				// check the inputs from the version in the blueprint
+				log.Printf("Error while getting catalog item inputs. Checking with blueprint instead")
+
+				if blueprintID != "" {
+					blueprintVersion := ""
+					if v, ok := d.GetOk("blueprint_version"); ok {
+						blueprintVersion = v.(string)
+					}
+
+					// Get the schema from blueprint to convert the provided input values
+					// to the type defined in the schema.
+					inputs, err = getBlueprintInputsByType(apiClient, blueprintID, blueprintVersion, v)
```

Comment:
The value of the `v`  parameter sent to the `getBlueprintInputsByType` function does not seem right. It should be the `inputs` value from L903, but you're assigning the `v` from L917, which corresponds to the value of the `blueprintVersion` attribute.

Reply:
I did a change to the code according to your feedback, and also removed the blueprintVersion. Instead, I used the catalogItemVersion, which should be the current one, from the already deployed item.